src/ast/passes/codegen_llvm.cpp

```diff
@@ -768,6 +768,15 @@ void CodegenLLVM::visit(Call &call)
       }
     }
     expr_ = nullptr;
+  } else if (call.func == "has_key") {
+    auto &arg = *call.vargs.at(0);
+    auto &map = static_cast<Map &>(arg);
+    auto [key, scoped_key_deleter] = getMapKey(map, call.vargs.at(1));
+
+    CallInst *lookup = b_.CreateMapLookup(map, key);
+    expr_ = b_.CreateICmpNE(b_.CreateIntCast(lookup, b_.GET_PTR_TY(), true),
+                            b_.GetNull(),
+                            "lookup_cond");
```

Comment:
Nit: a different name would make the expression easier to spot in codegen
```suggestion
                            "has_key");
```

Reply:
Good catch.